lib/attr_extras/attr_initialize/attributes_declaration.rb

```diff
@@ -0,0 +1,44 @@
+module AttrExtras
+  InvalidParameterDeclaration = Class.new(StandardError)
+
+  class AttrInitialize
+    class AttributesDeclaration
+      attr_reader :names_with_array_of_hash_arguments
+      private :names_with_array_of_hash_arguments
+
+      def initialize(names_with_array_of_hash_arguments)
+        @names_with_array_of_hash_arguments = names_with_array_of_hash_arguments
+      end
+
+      def positional_attributes
+        return @attributes_names if @attributes_names
+
+        @attributes_names = names_with_array_of_hash_arguments.take_while { |name_or_array| !name_or_array.is_a?(Array) }
+        raise InvalidParameterDeclaration if (names_with_array_of_hash_arguments - @attributes_names).count > 1
```

Comment:
Took me a little while to understand this (in the code), and I think the error will be cryptic if you get it. Maybe something along the lines of:

```
raise "Invalid parameter declaration! Must be 0 or more positional arguments followed by 0 or 1 array of hash arguments"
```

What do you think?

I can't think of any situation where anyone would need to rescue this error specifically, so I think I'd just go with a runtime error instead of a custom class: http://thepugautomatic.com/2015/01/exception-class/